src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/suppresswarnings/InputSuppressWarningsPatternVariables.java

```diff
@@ -0,0 +1,31 @@
+/*
+SuppressWarnings
+format = ^unchecked$*|^unused$*|.*
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF, \
+         ENUM_CONSTANT_DEF, PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, CTOR_DEF, \
+         COMPACT_CTOR_DEF, RECORD_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.annotation.suppresswarnings;
+
+public class InputSuppressWarningsPatternVariables {
+
+    void test(Object o) {
+        if (o instanceof ColoredPoint(@SuppressWarnings("")int x, int y, int z)) { } // violation
```

Comment:
@rnveach, @nrmancuso , when you see such lines without violation message, try to think a minute about "Should we actually demand message to be always specified?" (even Check has only one message)

I more and more inclined to think that it will be better to have violation message at all cases to let reader not rely on memory what it should be. In this case I completely forgot what this Check do, it required me jump  out of this page to get to know what this violation mean.

Preferable:
```
 // violation 'The warning '''' cannot be suppressed at this location.'
if (o instanceof ColoredPoint(@SuppressWarnings("")int x, int y, int z)) { }
```

Reply:
I see this as closed https://github.com/checkstyle/checkstyle/issues/11214 , and no issues on migration off `// violation` even for Checks with single message.
If you agree, please let me know, we will create issue it will be `good first/second issue`

Reply:
@nrmancuso has told me before to always specify the message so I think he agrees but I sometimes forgot to do so (There is no test that enforcs that). I will update this PR to specify the message. 
Ideally we should enfore the presence of the message even if the check has only one message and yes opening a good first issue to migrate all old `// violation`  to have message is a good idea

Reply:
Looks like I lost track that we finished migration to BDD, and now we can do this update.
Before it was completely ok to put violation message without message, as there were bunch missing useful variants violation marker were not implemented. But now we are good to go forward to use good practice format.

Reply:
We can enforce this in the parser itself or a simple regexp. There will be huge suppression lists

Reply:
@romani please create an issue on this so that we can discuss there

Reply:
https://github.com/checkstyle/checkstyle/pull/15452

Reply:
Follow up issue https://github.com/checkstyle/checkstyle/issues/15456